File: frontend/src/api/music.ts
import request from './request'
import type { PaginatedData } from '@/types/api'
import type {
  Song,
  SongCreateReq,
  SongUpdateReq,
  ParseTask,
  BatchCreateSongReq,
} from '@/types/music'

export const musicApi = {
  // 歌曲
  getSongs(params?: { category_id?: string; page?: number; page_size?: number }) {
    return request.get<PaginatedData<Song>>('/music/songs', { params })
  },
  createSong(data: SongCreateReq) {
    return request.post<Song>('/music/songs', data)
  },
  batchCreateSongs(data: BatchCreateSongReq) {
    return request.post<Song[]>('/music/songs/batch', data)
  },
  updateSong(id: string, data: SongUpdateReq) {
    return request.put<Song>(`/music/songs/${id}`, data)
  },
  deleteSong(id: string) {
    return request.delete(`/music/songs/${id}`)
  },

  // 解析
  parse(url: string) {
    return request.post<{ task_id: string }>('/music/parse', { url })
  },
  getParseStatus(taskId: string) {
    return request.get<ParseTask>(`/music/parse/${taskId}`)
  },

  // 获取播放地址（B站官方外链播放器地址，供 iframe 内嵌播放）
  getAudioUrl(songId: string) {
    return request.get<{ url: string }>(`/music/audio-url/${songId}`)
  },
}

File: backend/internal/router/music.go
package router

import (
	"novablog/internal/controller"

	"github.com/gin-gonic/gin"
)

// RegisterMusicRoutes 注册音乐播放器管理路由。
func RegisterMusicRoutes(r *gin.RouterGroup, musicController *controller.MusicController, authMiddleware gin.HandlerFunc) {
	music := r.Group("/music")
	music.Use(authMiddleware)
	{
		// 歌曲
		music.POST("/songs", musicController.CreateSong)
		music.POST("/songs/batch", musicController.BatchCreateSongs)
		music.GET("/songs", musicController.GetSongList)
		music.GET("/songs/:id", musicController.GetSongByID)
		music.PUT("/songs/:id", musicController.UpdateSong)
		music.DELETE("/songs/:id", musicController.DeleteSong)

		// 解析
		music.POST("/parse", musicController.ParseMusic)
		music.GET("/parse/:task_id", musicController.GetParseStatus)

		// 获取播放地址（返回B站官方外链播放器地址，前端用 iframe 内嵌播放）
		music.GET("/audio-url/:song_id", musicController.GetAudioURL)
	}
}
